### PR TITLE
fix: fixes clipboard detection on android when not installed

### LINKED ISSUE #1932: clipboard opt. dep. not detected correctly when not installed and causes a crash (Android only)
**Describe the bug**
On android, when `@react-native-clipboard/clipboard` is not installed, we give the user the option to copy a message that results in a crash.

**Dev environment info (please complete/provide the following information):**

- Package version: stream-chat-react-native v5.8.0 (in expo no issues 👍🏻)
- react-native: 0.68.0
- Device/Emulator (android/ios) and OS version: Android only

**To Reproduce**

* make sure `@react-native-clipboard/clipboard is not installed` in your app
1. Go to a channel
2. Long press on a message
3. Select "Copy text"
4. Crash occurs

**Expected behavior**
The option to copy the message should not be visible, and if by mistake it happens, the issue should never result in a crash.

**Additional context**
Detecting whether @react-native-clipboard/clipboard is installed does not work on android. This reason isn't clean and is obscure, but that might have to do with having it as a dev dependency in our package.

### PR DESCRIPTION
**Pull request by vanGalilea**
## 🛠 Implementation details

fixes #1932 

## 🎨 UI Changes
N/A

## 🧪 Testing

1. Run TSM app both on Android and iOS
2. Go to a random channel and long press on a text message
3. The option "copy text" should not be listed
4. Install the clipboard dependency in TSM app - `yarn add  @react-native-clipboard/clipboard`
5. Repeat step 2, the option "copy text" should be visible.
6. Click on "copy text" and the text should be available to paste somewhere

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [ ] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

File: package/native-package/package.json [hunk 1]
```diff
@@ -40,9 +40,7 @@
     "react-native-haptic-feedback": "^1.14.0",
     "react-native-image-crop-picker": "^0.38.0",
     "react-native-image-resizer": ">=1.4.2",
-    "react-native-share": ">=4.1.0"
-  },
-  "optionalDependencies": {
```
Comment (vanGalilea) [inline package/native-package/package.json:45]: `@react-native-clipboard/clipboard` under `optionalDependencies` caused a false positive detection when running `require("@react-native-clipboard/clipboard")`

Comment (vanGalilea) [reply package/native-package/package.json:45]: This has resulted in a crash

Comment (santhoshvai) [reply package/native-package/package.json:45]: This is not the right way.. the right way for using optional dependencies is like this

```
{
  "peerDependencies": {
    "@react-native-clipboard/clipboard": "^1.11.1"
  },
  "peerDependenciesMeta": {
    "@react-native-clipboard/clipboard": {
      "optional": true
    }
  }
}
```

ref: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta

**Review by santhoshvai — CHANGES_REQUESTED** [review]

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**